api/all/src/test/java/io/opentelemetry/api/baggage/propagation/W3CBaggagePropagatorTest.java

```diff
@@ -473,4 +473,34 @@ void inject_nullSetter() {
     W3CBaggagePropagator.getInstance().inject(context, carrier, null);
     assertThat(carrier).isEmpty();
   }
+
+  @Test
+  void shouldAllowBase64SpecialCharacters() {
+    String encodedInBase64 =
+        "MngJIFNQIAkhCSIJIwkkCSUJJgknCSgJKQkqCSsJLAktCS4JLwozeAkwCTEJMgkzCTQJNQk2CTcJOAk5CToJOwk8CT0JPgk/CjR4CUAJQQlCCUMJRAlFCUYJRwlICUkJSglLCUwJTQlOCU8KNXgJUAlRCVIJUwlUCVUJVglXCVgJWQlaCVsJXAldCV4JXwo2eAlgCWEJYgljCWQJZQlmCWcJaAlpCWoJawlsCW0JbglvCjd4CXAJcQlyCXMJdAl1CXYJdwl4CXkJegl7CXwJfQl+CQo4eAkJCQkJCQkJCQkJCQkJCQkKOXgJCQkJCQkJCQkJCQkJCQkJCkF4CU5CU1AJwqEJwqIJwqMJwqQJwqUJwqYJwqcJwqgJwqkJwqoJwqsJwqwJU0hZCcKuCcKvCkJ4CcKwCcKxCcKyCcKzCcK0CcK1CcK2CcK3CcK4CcK5CcK6CcK7CcK8CcK9CcK+CcK/CkN4CcOACcOBCcOCCcODCcOECcOFCcOGCcOHCcOICcOJCcOKCcOLCcOMCcONCcOOCcOPCkR4CcOQCcORCcOSCcOTCcOUCcOVCcOWCcOXCcOYCcOZCcOaCcObCcOcCcOdCcOeCcOfCkV4CcOgCcOhCcOiCcOjCcOkCcOlCcOmCcOnCcOoCcOpCcOqCcOrCcOsCcOtCcOuCcOvCkZ4CcOwCcOxCcOyCcOzCcO0CcO1CcO2CcO3CcO4CcO5CcO6CcO7CcO8CcO9CcO+CcO/Cgo=";
```

Comment:
On first glance this is a mysterious bit of code. Would prefer we trim it down to the sequence needed to replicate the issue, or modify `W3CBaggagePropagatorFuzzTest` to replicate the issue with randomness included. Here's an example of what that might look like. 

```
    @Fuzz
    public void baggageOctet(
        @From(BaggageOctetGenerator.class) String baggageValue) {
      Map<String, String> carrier = new HashMap<>();
      carrier.put("baggage", "key=" + baggageValue);
      Context context = baggagePropagator.extract(Context.current(), carrier, new MapTextMapGetter());
      Baggage baggage = Baggage.fromContext(context);
      String value = baggage.getEntryValue("key");
      assertThat(value).isEqualTo(baggageValue);
    }

  public static class BaggageOctetGenerator extends AbstractStringGenerator {
    
    private static final Set<Character> excluded = new HashSet<>(Arrays.asList(' ', '"', ',', ';', '\\', '%'));

    @Override
    protected int nextCodePoint(SourceOfRandomness random) {
      while (true) {
        char c = random.nextChar(' ', '~');
        if (!excluded.contains(c)) {
          return c;
        }
      }
    }

    @Override
    protected boolean codePointInRange(int codePoint) {
      return !excluded.contains((char) codePoint);
    }
  }
```

Reply:
Thanks, I'll clean it up and look into using of URLCodec.

Reply:
@jack-berg 
This fuzz test showed another issue. When baggage value starts with `=` sign, it is omitted, since it is treated as key/value separator. According to specification, there is only single `=` for separator, and baggage can start with it. 

So, something like this should be valid: `key====value`, resulting in `===value` value.

Am I understanding it correctly?

Reply:
Yup I came across that too! I came to the same conclusion. My solution was to add a new branch to a block in `Parser#parseInto(BaggageBuilder)`:
```
  void parseInto(BaggageBuilder baggageBuilder) {
    for (int i = 0, n = baggageHeader.length(); i < n; i++) {
      char current = baggageHeader.charAt(i);

      if (skipToNext) {
        if (current == ',') {
          reset(i + 1);
        }
        continue;
      }

      switch (current) {
        case '=':
          {
            if (state == State.KEY) {
              if (key.tryTerminating(i, baggageHeader)) {
                setState(State.VALUE, i + 1);
              } else {
                skipToNext = true;
              }
            } 
           // Start new code
           else if (state == State.VALUE) {
              skipToNext = !value.tryNextChar(current, i);
            }
           // End new code
```

Reply:
> This fuzz test showed another issue

💯

Reply:
Thanks for confirmation! 👍